### PR TITLE
Run Playwright tests in Node, not .NET

### LINKED ISSUE #346: Run node playwright tests as part of integration testing
This will give us more visibility into flakiness and failure patterns etc.

### PR DESCRIPTION
**Pull request by rmunn**
Fixes #346.

The .NET Playwright tests are going to become outdated, as new E2E tests will only be written in Typescript, not C#.

### CODE REVIEW

File: .github/workflows/integration-test.yaml [hunk 1]
```diff
@@ -93,7 +91,7 @@ jobs:
         shell: bash
         env:
           ZIP_PASSWORD: ${{ secrets.TEST_USER_PASSWORD }}
-        run: 7z a ./playwright-traces.7z ./bin/playwright-traces/*.* -p"$ZIP_PASSWORD"
+        run: 7z a ./playwright-traces.7z ./frontend/test-results/*/trace.zip -p"$ZIP_PASSWORD"
```
Comment (myieye) [inline .github/workflows/integration-test.yaml:94]: I'm pretty sure we want the whole `test-results` folder. The HTML-report is a big win that we didn't get in .NET.

File: .github/workflows/integration-test.yaml [hunk 2]
```diff
@@ -42,16 +42,6 @@ jobs:
     runs-on: ${{ inputs.runs-on }}
     steps:
       - uses: actions/checkout@v3
-      - name: Setup powershell
```
Comment (hahn-kev) [inline .github/workflows/integration-test.yaml:45]: please don't remove this, the setup python action requires powershell, also there's other non powershell stuff this is installing. It should be renamed to `Setup self hosted`

**Review by hahn-kev — CHANGES_REQUESTED** [review]
The setup powershell step was poorly named, but it is still required as other actions require powershell still.

File: .github/workflows/integration-test.yaml (hunk 1 above)
Comment (rmunn) [reply .github/workflows/integration-test.yaml]: Done in commit 0e4d407e.

File: .github/workflows/integration-test.yaml (hunk 2 above)
Comment (rmunn) [reply .github/workflows/integration-test.yaml:45]: Restored in commit 445b6432c8e9b997ed1e953cb0b5634ad775e65b, and renamed in commit d6042ea8474cf40d1994c9ec68df69d8b89203e9.

**Review by hahn-kev — CHANGES_REQUESTED** [review]

File: .github/workflows/integration-test.yaml [hunk 3]
```diff
@@ -71,15 +71,23 @@ jobs:
...
-          ${{inputs.runs-on != 'windows-latest' && 'sudo' || '' }} pwsh "$(dotnet msbuild --getProperty:OutputPath)playwright.ps1" install-deps
-          pwsh "$(dotnet msbuild --getProperty:OutputPath)playwright.ps1" install
+        working-directory: frontend
+        run: pnpm pretest
       - name: Build for tests
         run: "dotnet build /p:MercurialVersion=$MERCURIAL_VERSION"
         env:
           MERCURIAL_VERSION: ${{ inputs.hg-version }}
-      - name: Integration tests
+      - name: Integration tests (Playwright)
```
[the remark this answers is not in the document]
Comment (rmunn) [reply .github/workflows/integration-test.yaml:90]: Done in commit ce49e08e.

**Review by myieye — APPROVED** [review]
Looks OK to me. Eager to see what CI thinks 🤩

**Pull request merged**